Assets/LoggingManager/LoggingManager.cs

```diff
@@ -126,220 +95,147 @@ public void SaveLog(string collectionLabel)
         }
     }
 
+    public void SaveAllLogs()
+    {
+        foreach (KeyValuePair<string, LogStore> pair in logsList)
+        {
+            SaveLog(pair.Key);
+        }
+    }
+
+
     public void SetEmail(string newEmail)
     {
         email = newEmail;
     }
 
     public void CreateLog(string collectionLabel)
     {
-        collections.Add(collectionLabel, new LogCollection());
+        if (collectionLabel == "Meta")
+        {
+            AddMetaCollectionToList();
+            return;
+        }
+        LogStore logStore = new LogStore(collectionLabel, email, sessionID, logStringOverTime);
+        logsList.Add(collectionLabel, logStore);
```

Comment:
I am not sure we should be checking the `collectionLabel` here. Meta per se, is not supposed to be any kind of magic word. But, if `CreateMetaCollection` is enabled, LoggingManager will by default create a Meta log with DeviceID, SessionID, etc. What is this for exactly?

Reply:
The problem is that the code to log the Meta logs is "hard coded" in the program using the LoggingManager, so I don't really understand the purpose of the ```CreateMetaCollection``` boolean. 
Here is my pov:

>  if CreateMetaCollection is enabled, LoggingManager will by default create a Meta log with DeviceID, SessionID, etc. 

I agree, but if we want to add information to the Meta logs, we have to put these information in the program. For instance in the Arduino Logger, we add the Meta log data in the Arduino.cs class. 
In this case, we would have some lines like this :  
```c#
          LoggingManager.Log("Meta", new Dictionary<string, object>()
          {
              {"PID", pid},
              {"Comment", Comment},
          });
```

In order to add data to Meta logs, we must write the "Meta" string in the ```Log()``` function.
Now that this is written directly in the code, what if we put the ```CreateMetaCollection``` boolean to false ? 

What I wanted to do here is to allow the user to add data to Meta logs very easily while allowing Logging Manager to be compatible with the programs implementing its old version.

Maybe I misunderstood some things about how Meta logs should work in your environment ? 
should the user be able to add meta logs if he wants to, or are the Meta logs only there to log DeviceID, SessionID, etc and nothing else ?

Depending on the case, there should be several other ways of dealing with these meta logs.

Reply:
@Vincent-SD Thanks for your points. Here is an outline of how meta log behavior is supposed to work:

- When projects import LoggingManager, `CreateMetaCollection` is enabled by default. It means, that LoggingManager by default will create a `*Meta.csv` file, along with any other custom csv file they have logged.
- When `CreateMetaCollection` is enabled, LoggingManager will create a `LogStore` named `Meta` with `LogType.OneRowOverwrite` enabled and add some useful values like deviceID, a timestamp and a sessionID. This should ideally happen on `Start()`.
- Users of LoggingManager are free to write to the `Meta` collection as they please, to add custom values.
- `CreateMetaCollection` can be disabled, because in some cases, importing and merging many csv files in data analysis can be cumbersome. For example, people who use Excel for data analysis may want to just keep things simple and log everything into 1 CSV file. In that case, the `Meta` collection no longer exists and is in principle a name free to be used for other purposes.

To summarize, rather than checking for the name `Meta`, it is better to check for the boolean `CreateMetaCollection`.